game-core/src/main/java/games/strategy/triplea/settings/ClientSetting.java

```diff
@@ -43,88 +43,85 @@
-  public static final ClientSetting fasterArrowKeyScrollMultiplier =
-      new ClientSetting("FASTER_ARROW_KEY_SCROLL_MULTIPLIER", 2);
-  public static final ClientSetting spaceBarConfirmsCasualties =
-      new ClientSetting("SPACE_BAR_CONFIRMS_CASUALTIES", true);
-  public static final ClientSetting lobbyLastUsedHost = new ClientSetting("LOBBY_LAST_USED_HOST");
-  public static final ClientSetting lobbyLastUsedPort = new ClientSetting("LOBBY_LAST_USED_PORT");
-  public static final ClientSetting lookAndFeel =
-      new ClientSetting("LOOK_AND_FEEL_PREF", LookAndFeel.getDefaultLookAndFeelClassName());
-  public static final ClientSetting mapEdgeScrollSpeed = new ClientSetting("MAP_EDGE_SCROLL_SPEED", 30);
-  public static final ClientSetting mapEdgeScrollZoneSize = new ClientSetting("MAP_EDGE_SCROLL_ZONE_SIZE", 30);
-  public static final ClientSetting mapFolderOverride = new ClientSetting("MAP_FOLDER_OVERRIDE");
-  public static final ClientSetting mapListOverride = new ClientSetting("MAP_LIST_OVERRIDE");
-  public static final ClientSetting proxyChoice =
-      new ClientSetting("PROXY_CHOICE", HttpProxy.ProxyChoice.NONE.toString());
-  public static final ClientSetting proxyHost = new ClientSetting("PROXY_HOST");
-  public static final ClientSetting proxyPort = new ClientSetting("PROXY_PORT");
-  public static final ClientSetting saveGamesFolderPath =
-      new ClientSetting("SAVE_GAMES_FOLDER_PATH", new File(ClientFileSystemHelper.getUserRootFolder(), "savedGames"));
-  public static final ClientSetting serverObserverJoinWaitTime =
-      new ClientSetting("SERVER_OBSERVER_JOIN_WAIT_TIME", 180);
-  public static final ClientSetting serverStartGameSyncWaitTime =
-      new ClientSetting("SERVER_START_GAME_SYNC_WAIT_TIME", 180);
-  public static final ClientSetting showBattlesWhenObserving =
-      new ClientSetting("SHOW_BATTLES_WHEN_OBSERVING", true);
-  public static final ClientSetting showBetaFeatures = new ClientSetting("SHOW_BETA_FEATURES", false);
-  public static final ClientSetting showConsole = new ClientSetting("SHOW_CONSOLE", false);
-  public static final ClientSetting testLobbyHost = new ClientSetting("TEST_LOBBY_HOST");
-  public static final ClientSetting testLobbyPort = new ClientSetting("TEST_LOBBY_PORT");
-  public static final ClientSetting firstTimeThisVersion =
-      new ClientSetting("TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY", true);
-  public static final ClientSetting lastCheckForEngineUpdate =
-      new ClientSetting("TRIPLEA_LAST_CHECK_FOR_ENGINE_UPDATE");
-  public static final ClientSetting lastCheckForMapUpdates = new ClientSetting("TRIPLEA_LAST_CHECK_FOR_MAP_UPDATES");
-  public static final ClientSetting promptToDownloadTutorialMap =
-      new ClientSetting("TRIPLEA_PROMPT_TO_DOWNLOAD_TUTORIAL_MAP", true);
-  public static final ClientSetting userMapsFolderPath = new ClientSetting(
+public abstract class ClientSetting<T> implements GameSetting<T> {
```

Comment:
This class having all String types and the child class having a different type does not strike me as clean as it could be. Just my first impression, but I wonder about this. Still considering this update, I'm a bit on the fence.

Some questions:
- Are there any other types that we expect to be stored in ClientSetting?
- Should client setting be responsible for type specific parsing logic rather than just setting persistence? If the parsing logic were more complex, would that belong? We also invoke `HttpProxy.ProxyChoice`, which creates a new dependency in ClientSetting. The original design intended for ClientSetting to be a pretty low level persistence class.

Thinking about that last point, what would you think of flipping the dependency @ssoloff , instead of creating a child class instead use composition to have a ProxyClientSettings class that *uses* a client setting, essentially wrappers it to provide the API you are looking for?